addons/sale/report/invoice_report_templates.xml

```diff
@@ -1,21 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<odoo>
-    <template id="report_invoice_document_inherit_sale" inherit_id="account.report_invoice_document">
-        <xpath expr="//address" position="attributes">
-            <attribute name="groups">!sale.group_delivery_invoice_address</attribute>
```

Comment:
This xpath should also be moved
```xml
<address t-field="o.partner_id" t-options='{"widget": "contact", "fields": ["address", "name"], "no_marker": True}' />
```
to
```xml
<address t-field="o.partner_id"
         t-options='{"widget": "contact", "fields": ["address", "name"], "no_marker": True}'
         groups="!account.group_delivery_invoice_address"/>
```

Reply:
So here we're just adding the `groups` attribute to `o.partner_id` in `report_invoice.xml`, correct ?

Reply:
@casp-odoo In fact, the `address` block is there to convert a `res.partner` to a div containing all the address. So this xpath is just making the `address` block not displayed when you are using the `group_delivery_invoice_address` group.
The idea is to have 2 address blocks: one when you are using the shipping address, one when you are not. So you just need to add `groups="!account.group_delivery_invoice_address"` in `report_invoice.xml`